contrib/parseq-lambda-names/src/test/java/com/linkedin/parseq/lambda/TestUnrecognizedLambda.java

```diff
@@ -0,0 +1,110 @@
+package com.linkedin.parseq.lambda;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.function.Predicate;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+
+
+public class TestUnrecognizedLambda extends BaseTest {
+
+  private static final String CLASSNAME = TestUnrecognizedLambda.class.getSimpleName();
+
+  Optional<String> getDescriptionForVoidCallable(Callable<Void> c) {
+    return _asmBasedTaskDescriptor.getLambdaClassDescription(c.getClass().getName());
+  }
+
+  Optional<String> getDescriptionForIntCallable(Callable<Integer> c) {
+    return _asmBasedTaskDescriptor.getLambdaClassDescription(c.getClass().getName());
+  }
+
+  Optional<String> getDescriptionForStringPredicate(Predicate<String> c) {
+    return _asmBasedTaskDescriptor.getLambdaClassDescription(c.getClass().getName());
+  }
+
+  @Test
+  public void testReturnExpression() {
+    Optional<String> description = getDescriptionForCallable(() -> {return "";});
+    assertTrue(description.isPresent());
+    assertNameMatch("", "testReturnExpression", CLASSNAME, description.get().toString());
+  }
+
+  @Test
+  public void testExpressions() {
+    Optional<String> description = getDescriptionForVoidCallable(() -> {
+      int a = 5;
+      int b = 10;
+      int c = a + b;
+      return null;
+    });
+    assertTrue(description.isPresent());
+    assertNameMatch("", "testExpressions", CLASSNAME, description.get().toString());
+  }
+
+  @Test
+  public void testOperations() {
+    MathOperation multiplication = (int a, int b) -> { return a * b; };
+    Optional<String> description = getDescriptionForIntCallable(() -> {return this.operate(5, 3, multiplication);});;
+    assertTrue(description.isPresent());
+    //if operate function's return type is changed to Integer, we aren't able to infer operation
+    assertNameMatch("operate(_,_,_).Integer.valueOf(_)", "testOperations", CLASSNAME, description.get().toString());
```

Comment:
This description looks weird: `operate(_,_,_).Integer.valueOf(_)`
There are two methods `operate(_,_,_)` and `Integer.valueOf(_)` and they are connected with `.`.

Maybe we could detect autoboxing and truncate it from description? I think it would be reasonable to truncate Integer.valueOf(int) if return type of a lambda is Integer. Similar for other primitives e.g. truncate Long.valueOf(long) if return type of a lambda is Long

This way description would be `operate(_,_,_)`.